controllers/datafoundationclient.go

```diff
@@ -0,0 +1,217 @@
+package controllers
+
+import (
+	"fmt"
+	"net/url"
+	"strings"
+
+	opv1a1 "github.com/operator-framework/api/pkg/operators/v1alpha1"
+	v1alpha1 "github.com/red-hat-storage/managed-fusion-agent/api/v1alpha1"
+	ocsclient "github.com/red-hat-storage/ocs-client-operator/api/v1alpha1"
+	"k8s.io/apimachinery/pkg/runtime"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+)
+
+const (
+	ocsClientOperatorName = "ocs-client-operator"
+)
+
+type dataFoundationClientSpec struct {
+	onboardingTicket string
+	providerEndpoint string
+}
+
+type dataFoundationClientReconciler struct {
+	*ManagedFusionOfferingReconciler
+
+	offering                      *v1alpha1.ManagedFusionOffering
+	dataFoundationClientSpec      dataFoundationClientSpec
+	storageClient                 ocsclient.StorageClient
+	defaultBlockStorageClassClaim ocsclient.StorageClassClaim
+	defaultFSStorageClassClaim    ocsclient.StorageClassClaim
+}
+
+//+kubebuilder:rbac:groups=ocs.openshift.io,resources=storageclients,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=ocs.openshift.io,resources=storageclassclaims,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=operators.coreos.com,namespace=system,resources=clusterserviceversions,verbs=get;list;watch;delete;update;patch
+
+func dfcSetupWatches(controllerBuilder *builder.Builder) {
+	csvPredicates := builder.WithPredicates(
+		predicate.NewPredicateFuncs(
+			func(res client.Object) bool {
+				return strings.HasPrefix(res.GetName(), ocsClientOperatorName)
+			},
+		),
+	)
+	controllerBuilder.
+		Owns(&opv1a1.ClusterServiceVersion{}, csvPredicates).
+		Owns(&ocsclient.StorageClient{}).
+		Owns(&ocsclient.StorageClassClaim{})
+}
+
+func DFCAddToScheme(scheme *runtime.Scheme) {
+	utilruntime.Must(ocsclient.AddToScheme(scheme))
+}
+
+func dfcReconcile(offeringReconciler *ManagedFusionOfferingReconciler, offering *v1alpha1.ManagedFusionOffering) (ctrl.Result, error) {
+	r := dataFoundationClientReconciler{}
+	r.initReconciler(offeringReconciler, offering)
+
+	if err := r.parseSpec(offering); err != nil {
+		return ctrl.Result{}, err
+	}
+	return r.reconcilePhases()
+}
+
+func (r *dataFoundationClientReconciler) initReconciler(reconciler *ManagedFusionOfferingReconciler, offering *v1alpha1.ManagedFusionOffering) {
+	r.ManagedFusionOfferingReconciler = reconciler
+	r.offering = offering
+
+	r.storageClient.Name = "ocs-storageclient"
+	r.storageClient.Namespace = offering.Namespace
+
+	r.defaultBlockStorageClassClaim.Name = "ocs-storagecluster-ceph-rbd"
+	r.defaultFSStorageClassClaim.Name = "ocs-storagecluster-cephfs"
```

Comment:
To keep this function consistent, let's leave an empty line between int code for different resources
```suggestion
	r.defaultBlockStorageClassClaim.Name = "ocs-storagecluster-ceph-rbd"
	
	r.defaultFSStorageClassClaim.Name = "ocs-storagecluster-cephfs"
```

Reply:
updated